src/controller/busControllers.js

```diff
@@ -205,6 +205,96 @@ class busController {
         }
 
     }
+
+    static async assignRoute(req, res){
+        try {
+            const findRoute = await Route.findOne({
+                where:{ routeID: req.body.routeID }
+            })
+
+            if(!findRoute){
+                return res.status(404).json({
+                    message: res.__("route doesn't exist")
+                })
+            }
+    
+            const updated = await Bus.update({ routeId:findRoute.id },
+                { where: { id: req.params.id } });
+
+            if (updated) {
+                const assignedBus= await Bus.findOne({
+                    where: { id:req.params.id },
+                    include: [{
+                        model: Route,
+                        as: 'route',
+                        attributes:["id","name","routeID"]
+                    }]
```

Comment:
don't fetch again
use updated to return data to be returned


UPDATE: get one bus functionality to include the Route model associated with it